File: python/hackerrank/delete-duplicate-value-nodes-from-a-sorted-linked-list.py
"""
 Delete duplicate nodes
 head could be None as well for empty list
 Node is defined as
 
 class Node(object):
 
   def __init__(self, data=None, next_node=None):
       self.data = data
       self.next = next_node

 return back the head of the linked list in the below method.
"""

def RemoveDuplicates(head):
		root_node = Node(0)
		root_node.next = head
		prev_head = root_node
		last_val = None
		while prev_head.next is not None:
			if prev_head.next.data == last_val:
				last_val = prev_head.next.data
				prev_head.next = prev_head.next.next
			else:
				last_val = prev_head.next.data
				prev_head = prev_head.next
		return root_node.next

File: javascript/hackerrank/delete-duplicate-value-nodes-from-a-sorted-linked-list.js
/*
    Remove all duplicate elements from a sorted Linked List
    Node is defined as
    var Node = function(data) {
        this.data = data;
        this.next = null;
    }
*/

// This is a "method-only" submission.
// You only need to complete this method.

function removeDuplicates(head) {
    rootNode = new Node(0)
    rootNode.next = head
    prevHead = rootNode
    var lastVal
    while (prevHead.next !== null) {
        if (prevHead.next.data == lastVal) {
            lastVal = prevHead.next.data
            prevHead.next = prevHead.next.next
        } else {
            lastVal = prevHead.next.data
            prevHead = prevHead.next
        }
    }
    return rootNode.next
}